### PR TITLE
[RISCV] Disable generation of asynchronous unwind tables for RISCV baremetal

### PR DESCRIPTION
**Pull request by quic-garvgupt**
The below culprit patch enabled the generation of asynchronous unwind tables (-funwind-tables=2) by default for RISCV for both linux and RISCVToolChain baremetal object. However, since there are 2 baremetal toolchain objects for RISCV, this created a discrepancy between their behavior. Moreover, enabling the generation of asynchronous unwind tables based on whether `-gcc-toolchain` option is present or not doesn't seem to be the best criteria to decide on the same. This patch make the behavior consistent by disabling the unwind tables in RISCVToolChain Baremetal object.

Culprit Patch - https://reviews.llvm.org/D145164

### CODE REVIEW

**Comment by quic-garvgupt** [comment]
Hi
@jonathonpenix  @apazos @kito-cheng @asb @MaskRay - Requesting for adding as reviewers to review the PR. Thanks in advance!

**Comment by MaskRay** [comment]
I wonder whether GCC considers using `-fno-asynchronous-unwind-tables` for all RISC-V configurations.

**Comment by kito-cheng** [comment]
RISC-V GCC has enabled `-fasynchronous-unwind-tables` and `-funwind-tables` by default for Linux target, and disabled by default for baremetal, so generally LGTM since it align the behavior with GCC, but I would like to wait @asb's response. 

NOTE: The patch[1] is come from SUSE folks, so I assume it's necessary from the distro's point of view.

Ref:
[1] https://github.com/gcc-mirror/gcc/commit/3cd08f7168c196d7a481b9ed9f4289fd1f14eea8

**Comment by asb** [comment]
CC @petrhosek too who I know [hopes to merge the RISCVToolChain and BareMetal toolchains](https://discourse.llvm.org/t/merging-riscvtoolchain-and-baremetal-toolchains/75524). Matching the GCC behaviour seems sensible to me, and I'm hopeful that the planned future refactoring will avoid this kind of discrepancy in behaviour based on `--gcc-toolchain` presence in the future.

**Comment by quic-garvgupt** [comment]
Hi @asb, since Petr already gave a good-ahead to this patch in the meeting, let me know if we can merge this PR if there are no new changes need to be made

**Review by petrhosek — APPROVED** [review]

File: clang/test/Driver/clang-translation.c
Comment (MaskRay) [inline clang/test/Driver/clang-translation.c:425]: clang/test/Driver/riscv-features.c seems better. clang-translation.c tests miscellaneous stuff.

**Review by MaskRay — APPROVED** [review]

File: clang/test/Driver/clang-translation.c
Comment (quic-garvgupt) [reply clang/test/Driver/clang-translation.c]: Done

**Comment by quic-garvgupt** [comment]
Hi @asb, I do not have write access,  requesting to merge this PR on my behalf. Also, requesting to backport this PR onto the release  branch. Thanks in advance!

**Comment by quic-garvgupt** [comment]
Hi @MaskRay, please review the updated changes and if they are fine, please merge this PR on my behalf. Thanks!

**Comment by quic-garvgupt** [comment]
ping! for merging this PR

**Pull request merged**
